crates/sui-types/src/event.rs

```diff
@@ -22,19 +23,27 @@ use crate::{
 #[derive(Debug, Clone, PartialEq)]
 pub struct EventEnvelope {
     /// UTC timestamp in milliseconds since epoch (1/1/1970)
-    timestamp: u64,
+    timestamp: u128,
     /// Transaction digest of associated transaction, if any
     tx_digest: Option<TransactionDigest>,
     /// Specific event type
-    event: Event,
+    pub event: Event,
+    /// JSON contents for MoveEvent, if present
+    pub move_event_json_contents: Option<Value>,
 }
 
 impl EventEnvelope {
-    pub fn new(timestamp: u64, tx_digest: Option<TransactionDigest>, event: Event) -> Self {
+    pub fn new(
+        timestamp: u128,
+        tx_digest: Option<TransactionDigest>,
+        event: Event,
+        move_event_json_contents: Option<Value>,
```

Comment:
I don't think the move_event_json_contents thing should be pre-populated the first time.... there are better ways to do this as it might be expensive.  We can use OnceCell, so that it's lazy, OR better yet maybe we can get Move to give us the original MoveStruct without deserializing it.

Reply:
> Yeah I can include MoveStruct here. But we probably still need move_event_json_contents as it's useful for subscription filtering. I will take another look to confirm though.

Two things:
1. It's more efficient to filter using MoveStruct than JSON representation, in particular if you want to filter by things like ObjectID, Address, etc. because doing binary comparisons is much cheaper than the string (or hex representation) comparisons.
2. The primary fields we are trying to filter by, including the package and module ID, and event type, are not in the MoveStruct that is deserialized, but rather from the MoveLayout.  In fact those are fields that are in other event types too, not just the user MoveEvents.  

2 is the reason that I say, maybe the move struct / json stuff should be optional.

Also, move struct -> JSON is very cheap, the expensive part is deserializing raw bytes into a MoveStruct, I'm pretty sure.   Well, I guess we can measure it.  

I'm also suggesting to standardize on MoveStruct because from a storage perspective, it is better to store something closer to the original structure, for example preserving binary types and blobs.  The JSON representation is not as canonical, like there is for example no canonical way to represent binary data in JSON.  That said there are advantages to JSON as well,

Reply:
resolved offline.

FWIW, json is more filtering friendly because we can easily say "json_val['foo']['bar']". For MoveStruct we have to invent the wheels by ourselves, which I would punt unless there's a strong incentive to do so